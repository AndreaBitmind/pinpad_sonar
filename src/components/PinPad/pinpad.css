.pinpad-container {
  border: 1px solid black;
  border-radius: 15px;
  max-width: 600px;
  height: 800px;
  display: flex;
  padding: 47px;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
