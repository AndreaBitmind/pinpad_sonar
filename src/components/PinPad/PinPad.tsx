import { useState, useEffect } from "react";
import Button from "../Button/Button";
import Display from "../Display/Display";
import "./pinpad.css";

const pinpadNumbers = [7, 8, 9, 4, 5, 6, 3, 2, 1, 0];
const pin = ["8", "5", "3", "0"];
const initialState: string[] = [];
const messageOK = ["OK"];
const messageERROR = ["ERROR"];

const PinPad = (): JSX.Element => {
  const [result, setResult] = useState(initialState);

  const handleChange = (digit: number) => {
    if (result.length < 4) {
      setResult([...result, digit.toString()]);
    }
  };

  useEffect(() => {
    if (result.length === 4) {
      if (pin.toString() === result.toString()) {
        setTimeout(() => setResult(messageOK), 1000);
      } else {
        setTimeout(() => setResult(messageERROR), 1000);
      }

      setTimeout(() => setResult([]), 2000);
    }
  }, [result]);

  const digits = pinpadNumbers.map((number) => {
    return (
      <li key={number}>
        <Button digit={number} handleChange={() => handleChange(number)} />
      </li>
    );
  });

  return (
    <div className="pinpad-container">
      <Display result={result} />
      <ul className="pinpad">{digits}</ul>
    </div>
  );
};

export default PinPad;
