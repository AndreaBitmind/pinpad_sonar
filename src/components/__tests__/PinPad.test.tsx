import TestRenderer from "react-test-renderer";
import PinPad from "../PinPad/PinPad";

describe("Given a PinPad component", () => {
  describe("When it's rendered in the screen", () => {
    test("Then it should show the PinPad component", () => {
      const expectedPinPadComponent = TestRenderer.create(<PinPad />);

      expect(expectedPinPadComponent).toMatchSnapshot();
    });
  });
});
