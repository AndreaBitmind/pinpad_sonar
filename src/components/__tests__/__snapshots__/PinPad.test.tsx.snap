// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a PinPad component When it's rendered in the screen Then it should show the PinPad component 1`] = `
<div
  className="pinpad-container"
>
  <div
    className="pinpad-container_header"
  />
  <div
    className="display"
  />
  <ul
    className="pinpad"
  >
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        7
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        8
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        9
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        4
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        5
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        6
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        3
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        2
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        1
      </button>
    </li>
    <li>
      <button
        className="button"
        onClick={[Function]}
      >
        0
      </button>
    </li>
  </ul>
</div>
`;
